### PR TITLE
Add melee_training_cap into monsters.md

### PR DESCRIPTION
**Pull request by GuardianDll**
#### Summary
None
#### Purpose of change
melee_training_cap is not documented anywhere 
#### Describe the solution
Document `melee_training_cap` in our docs

### CODE REVIEW

File: doc/MONSTERS.md [hunk 1]
```diff
@@ -105,6 +105,7 @@ Monsters may also have any of these optional properties:
 | `absorb_move_cost_max`   | (int) For monsters with the `ABSORB_ITEMS` special attack. Sets a maximum movement cost for absorbing items regardless of the volume of the consumed item. -1 for no limit. Default -1.
 | `absorb_material`        | (array of string) For monsters with the `ABSORB_ITEMS` special attack. Specifies the types of materials that the monster will seek to absorb. Items with multiple materials will be matched as long as it is made of at least one of the materials in this list. If not specified the monster will absorb all materials.
 | `split_move_cost`        | (int) For monsters with the `SPLIT` special attack. Determines the move cost when splitting into a copy of itself.
+| `melee_training_cap `    | (int) how much you can train your melee skill by attack this monster. 10 by default.
```
Comment (GuardianDll) [inline doc/MONSTERS.md:108]:
```suggestion
| `melee_training_cap `    | (int) How much you can train your melee skill by attacking this monster. 10 by default.
```

**Comment by Karol1223** [comment]
You mentioned breathers in your issue regarding this, I suppose you're leaving them out on purpose?

**Comment by GuardianDll** [comment]
They have melee_training_cap 2 already

**Review by Maleclypse — APPROVED** [review]
Thank you for documenting this!

**Comment by Fris0uman** [comment]
Why not train melee on person though?

**Comment by Fris0uman** [comment]
So like a dummy basically, so it should at least train to the level a dummy does

**Comment by GuardianDll** [comment]
So to level 4 you mean?

**Comment by Fris0uman** [comment]
I think it's weird to have a property to arbitrary cap melee training, this suggest that there's something wrong in how melee training works and that this solution just paper over it instead of fixing it. if melee training means that you're fighting something that is moving and dodging around then it should scale with the dodging ability of the monster for exemple and person should have none.

File: data/json/monsters/nether.json [hunk 2]
```diff
@@ -1345,6 +1345,7 @@
     "vision_night": 60,
     "path_settings": { "max_dist": 60 },
     "death_function": { "corpse_type": "NO_CORPSE", "message": "it just disappeared." },
+    "melee_training_cap": 0,
```
Comment (GuardianDll) [inline data/json/monsters/nether.json:1348]:
```suggestion
```

**Comment by GuardianDll** [comment]
Changed the PR prior to #63773

**Comment by GuardianDll** [comment]
oh wait, it already add the documentation info. Nevermind then

**Pull request closed without merging**